stratum/hal/lib/barefoot/bf_switch.cc

```diff
@@ -101,9 +105,58 @@ ::util::Status BFSwitch::VerifyChassisConfig(const ChassisConfig& config) {
   return status;
 }
 
+namespace {
+std::string Uint32ToLeByteStream(uint32 val) {
+  uint32 tmp = (htonl(1) == 1) ? __builtin_bswap32(val) : val;
+  std::string bytes = "";
+  bytes.assign(reinterpret_cast<char*>(&tmp), sizeof(uint32));
+  return bytes;
+}
+
+::util::Status BfDeviceConfigToPiNodeConfig(const BfDeviceConfig& device_config,
+                                            std::string* pi_node_config) {
+  CHECK_RETURN_IF_FALSE(pi_node_config) << "null pointer.";
+
+  // Validate restrictions.
+  CHECK_RETURN_IF_FALSE(device_config.programs_size() == 1)
+      << "Only single program P4 configs are supported.";
+  const auto& program = device_config.programs(0);
+  CHECK_RETURN_IF_FALSE(program.pipelines_size() == 1)
+      << "Only single pipeline P4 configs are supported.";
+  const auto& pipeline = program.pipelines(0);
+
+  pi_node_config->clear();
+  // Program name
+  pi_node_config->append(Uint32ToLeByteStream(program.name().size()));
+  pi_node_config->append(program.name());
+  // Tofino bin
+  pi_node_config->append(Uint32ToLeByteStream(pipeline.config().size()));
+  pi_node_config->append(pipeline.config());
+  // Context json
+  pi_node_config->append(Uint32ToLeByteStream(pipeline.context().size()));
+  pi_node_config->append(pipeline.context());
+  VLOG(1) << "PI node config: " << StringToHex(*pi_node_config);
```

Comment:
This could be megabytes of log, so maybe we don't want to do this?

Reply:
Debugging leftover